test/integration_test.go

```diff
@@ -6351,68 +6326,41 @@ func (ts *IntegrationTestSuite) TestVersioningBehaviorPerWorkflowType() {
 	for i := 1; i < len(versioningBehaviorAll); i++ {
 		ts.Equal(versioningBehaviorAll[i], enumspb.VERSIONING_BEHAVIOR_PINNED)
 	}
-
-	// Show that an explicit setter overrides workflow type default
-	versioningBehaviorAll = nil
-
-	wfOpts = ts.startWorkflowOptions("test-override-versioning-behavior-per-type")
-	ts.NoError(ts.executeWorkflowWithOption(wfOpts,
-		ts.workflows.SetPinnedVersioningBehavior, nil))
-
-	ts.Equal(enumspb.VERSIONING_BEHAVIOR_PINNED, versioningBehaviorAll[0])
-	for i := 1; i < len(versioningBehaviorAll); i++ {
-		ts.Equal(versioningBehaviorAll[i], enumspb.VERSIONING_BEHAVIOR_PINNED)
-	}
 }
 
-func (ts *IntegrationTestSuite) TestGetVersioningBehavior() {
-	ctx, cancel := context.WithTimeout(context.Background(), ctxTimeout)
-	defer cancel()
-
-	// We are setting the default build ID with versioning-2 rules to test
-	// with existing servers. TODO(antlai-temporal) use versioning-3 APIs
-	// after there is a server release that supports versioning-3
-	res, err := ts.client.GetWorkerVersioningRules(ctx, client.GetWorkerVersioningOptions{
-		TaskQueue: ts.taskQueueName,
-	})
-	ts.NoError(err)
-
-	_, err = ts.client.UpdateWorkerVersioningRules(ctx, client.UpdateWorkerVersioningRulesOptions{
-		TaskQueue:     ts.taskQueueName,
-		ConflictToken: res.ConflictToken,
-		Operation: &client.VersioningOperationInsertAssignmentRule{
-			RuleIndex: 0,
-			Rule: client.VersioningAssignmentRule{
-				TargetBuildID: "1.0",
-			},
+func (ts *IntegrationTestSuite) TestNoVersioningBehaviorPanics() {
+	c, err := client.Dial(client.Options{
+		HostPort:  ts.config.ServiceAddr,
+		Namespace: ts.config.Namespace,
+		ConnectionOptions: client.ConnectionOptions{
+			TLS: ts.config.TLS,
 		},
 	})
 	ts.NoError(err)
+	defer c.Close()
 
 	ts.worker.Stop()
 	ts.workerStopped = true
-	w := worker.New(ts.client, ts.taskQueueName, worker.Options{
+	w := worker.New(c, ts.taskQueueName, worker.Options{
 		BuildID:                 "1.0",
 		UseBuildIDForVersioning: true,
```

Comment:
is there any different between `UseBuildIDForVersioning` and `DeploymentOptions.UseBuildIDForVersioning`? Do we want to make `UseBuildIDForVersioning` exclusively for versioning 1-2 and the latter for versioning 3?

Reply:
There is no `DeploymentOptions.UseBuildIDForVersioning` until we deprecate versioning-2, otherwise it is just too confusing to have same field twice without deprecation.